File: src/test/resources/languages/cpp/comprehensive.cpp
#include <iostream>
#include <vector>
#include <algorithm>

/**
 * Block comment for namespace.
 */
namespace extractor {

    // Single line comment
    template <typename T>
    class Comprehensive {
    private:
        T value;

    public:
        Comprehensive(T v) : value(v) {}

        /**
         * Method with lambda inside.
         */
        void process() {
            auto lambda = [this](int x) {
                return this->value + x;
            };

            std::vector<int> vec = {1, 2, 3};
            std::for_each(vec.begin(), vec.end(), [](int& n) {
                n *= 2;
            });
        }

        // Static method
        static void info() {
            std::cout << "Info" << std::endl;
        }
    };

} // namespace extractor

int main() {
    extractor::Comprehensive<int> c(10);
    c.process();
    return 0;
}

File: src/test/resources/languages/java/comprehensive.java
package com.extractor.comprehensive;

import java.util.List;
import java.util.stream.Collectors;
import java.util.Arrays;

/**
 * Block comment at the top of the file.
 */
@Deprecated
@SuppressWarnings("unchecked")
public class Comprehensive<T> {

    // Single-line comment for a field
    private List<T> items;

    /**
     * Constructor block comment.
     */
    public Comprehensive() {
        this.items = (List<T>) new java.util.ArrayList<>();
    }

    /*
     * Block comment for a method
     */
    @Override
    @Deprecated
    public String toString() {
        return "Comprehensive{" +
                "items=" + items +
                '}';
    }

    public void processItems() {
        // Lambda expression
        items.forEach(item -> {
            System.out.println(item);
        });

        // Stream API
        List<String> results = items.stream()
                .filter(i -> i != null)
                .map(Object::toString)
                .collect(Collectors.toList());
    }

    @interface CustomAnnotation {
        String value() default "";
    }

    public static class InnerClass {
        public void innerMethod() {
            // Nested block
            {
                int x = 10;
            }
        }
    }

    static {
        // Static initialization block
        System.out.println("Static block");
    }
}
